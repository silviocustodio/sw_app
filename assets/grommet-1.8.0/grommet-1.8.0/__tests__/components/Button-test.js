// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react-test-renderer';

import Button from '../../src/js/components/Button';

import FakeIcon from '../mocks/FakeIcon';

describe('Button', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Button label='Test me' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct primary={true} rendering', () => {
    const component = renderer.create(
      <Button label='Test me' primary={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct plain={true} rendering', () => {
    const component = renderer.create(
      <Button label='Test me' plain={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct icon rendering', () => {
    const component = renderer.create(
      <Button icon={<FakeIcon />} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct icon as children rendering', () => {
    const component = renderer.create(
      <Button><FakeIcon /></Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has text as children rendering', () => {
    const component = renderer.create(
      <Button>test</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has className property rendering', () => {
    const component = renderer.create(
      <Button className="test">test</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has microdata properties rendering', () => {
    const component = renderer.create(
      <Button itemScope={true} itemType="http://schema.org/Article"
        itemProp="test">test</Button>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
