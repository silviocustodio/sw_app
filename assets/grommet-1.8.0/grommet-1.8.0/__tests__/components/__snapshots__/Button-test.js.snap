// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button has className property rendering 1`] = `
<button
  aria-label={undefined}
  className="grommetux-button grommetux-button--disabled grommetux-button--plain test"
  disabled={true}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  test
</button>
`;

exports[`Button has correct default options 1`] = `
<button
  aria-label={undefined}
  className="grommetux-button grommetux-button--disabled"
  disabled={true}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <span
    className="grommetux-button__label"
  >
    Test me
  </span>
</button>
`;

exports[`Button has correct icon as children rendering 1`] = `
<button
  aria-label={undefined}
  className="grommetux-button grommetux-button--disabled grommetux-button--plain"
  disabled={true}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <svg
    height="24px"
    role="img"
    version="1.1"
    viewBox="0 0 24 24.3616"
    width="24px"
  />
</button>
`;

exports[`Button has correct icon rendering 1`] = `
<button
  aria-label={undefined}
  className="grommetux-button grommetux-button--disabled grommetux-button--plain"
  disabled={true}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <span
    className="grommetux-button__icon"
  >
    <svg
      height="24px"
      role="img"
      version="1.1"
      viewBox="0 0 24 24.3616"
      width="24px"
    />
  </span>
</button>
`;

exports[`Button has correct plain={true} rendering 1`] = `
<button
  aria-label={undefined}
  className="grommetux-button grommetux-button--disabled grommetux-button--plain"
  disabled={true}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <span
    className="grommetux-button__label"
  >
    Test me
  </span>
</button>
`;

exports[`Button has correct primary={true} rendering 1`] = `
<button
  aria-label={undefined}
  className="grommetux-button grommetux-button--primary grommetux-button--disabled"
  disabled={true}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  <span
    className="grommetux-button__label"
  >
    Test me
  </span>
</button>
`;

exports[`Button has microdata properties rendering 1`] = `
<button
  aria-label={undefined}
  className="grommetux-button grommetux-button--disabled grommetux-button--plain"
  disabled={true}
  href={undefined}
  itemProp="test"
  itemScope={true}
  itemType="http://schema.org/Article"
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  test
</button>
`;

exports[`Button has text as children rendering 1`] = `
<button
  aria-label={undefined}
  className="grommetux-button grommetux-button--disabled grommetux-button--plain"
  disabled={true}
  href={undefined}
  onBlur={[Function]}
  onClick={undefined}
  onFocus={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  type="button"
>
  test
</button>
`;
